docusaurus/docs/how-to-guides/app-plugins/add-resource-handler.md

```diff
@@ -0,0 +1,265 @@
+
+	_, err := rw.Write([]byte(`{ "namespaces": ["ns-1", "ns-2"] }`))
+	if err != nil {
+		ctxLogger.Error("Failed to write response", "error", err)
+		return
+	}
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (a *MyApp) handleProjects(rw http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodGet {
+		return
+	}
+
+	ctxLogger := backend.Logger.FromContext(req.Context())
+
+	_, err := rw.Write([]byte(`{ "projects": ["project-1", "project-2"] }`))
+	if err != nil {
+		ctxLogger.Error("Failed to write response", "error", err)
+		return
+	}
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (a *MyApp) updateDevice(rw http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		return
+	}
+
+	if req.Body == nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	ctxLogger := backend.Logger.FromContext(req.Context())
+
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			ctxLogger.Warn("Failed to close response body", "error", err)
+		}
+	}()
+
+	var payload map[string]any
+	b, err := io.ReadAll(req.Body)
+	if err != nil {
+		ctxLogger.Error("Failed to read request body to bytes", "error", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	err := json.Unmarshal(b, &payload)
+	if err != nil {
+		ctxLogger.Error("Failed to unmarshal request body to JSON", "error", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	state := payload["state"]
+	// update device with state...
+
+	_, err := rw.Write([]byte(`{ "message": "device updated"] }`))
+	if err != nil {
+		ctxLogger.Error("Failed to write response", "error", err)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+```
+
+:::note
+
+Using some other HTTP router library with above example should be straightforward. Just replace the use of [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) with another router.
+
+:::
+
+### What if you need access to the backend plugin context?
+
+Use the `backend.PluginConfigFromContext` function to access `backend.PluginContext`:
+
+```go
+func (a *MyApp) handleNamespaces(rw http.ResponseWriter, req *http.Request) {
+	pCtx := backend.PluginConfigFromContext(req.Context())
+	ctxLogger := backend.Logger.FromContext(req.Context())
+
+	bytes, err := json.Marshal(pCtx.User)
+	if err != nil {
+		ctxLogger.Error("Failed to marshal user to JSON bytes", "error", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	_, err := rw.Write(bytes)
+	if err != nil {
+		ctxLogger.Error("Failed to write response", "error", err)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+```
+
+## Accessing app resources from the frontend
+
+You can query your resources using the `getResource` and `postResource` helpers from the `DataSourceWithBackend` class. To provide a nicer and more convenient API for your components it's recommended to extend your datasource class and instance with functions for each route as shown in the following example:
```

Comment:
This content is about data sources, but this guide is about app plugins so it won't work.